src/openrct2-ui/windows/ClearScenery.cpp

```diff
@@ -33,208 +33,169 @@ static constexpr const rct_string_id WINDOW_TITLE = STR_CLEAR_SCENERY;
-    window->hold_down_widgets = (1ULL << WIDX_INCREMENT) | (1ULL << WIDX_DECREMENT);
-    WindowInitScrollWidgets(window);
-    window_push_others_below(window);
-
-    gLandToolSize = 2;
-    gClearSceneryCost = MONEY64_UNDEFINED;
-
-    gClearSmallScenery = true;
-    gClearLargeScenery = false;
-    gClearFootpath = false;
-
-    return window;
-}
-
-/**
- *
- *  rct2: 0x006E6B65
- */
-static void window_clear_scenery_close([[maybe_unused]] rct_window* w)
-{
-    // If the tool wasn't changed, turn tool off
-    if (clear_scenery_tool_is_active())
-        tool_cancel();
-}
-
-/**
- *
- *  rct2: 0x0068E185
- */
-static void window_clear_scenery_mouseup(rct_window* w, rct_widgetindex widgetIndex)
-{
-    switch (widgetIndex)
-    {
-        case WIDX_CLOSE:
-            window_close(w);
-            break;
-        case WIDX_PREVIEW:
-            window_clear_scenery_inputsize(w);
-            break;
-        case WIDX_SMALL_SCENERY:
-            gClearSmallScenery ^= 1;
-            w->Invalidate();
-            break;
-        case WIDX_LARGE_SCENERY:
-            gClearLargeScenery ^= 1;
-            w->Invalidate();
-            break;
-        case WIDX_FOOTPATH:
-            gClearFootpath ^= 1;
-            w->Invalidate();
-            break;
-    }
-}
-
-static void window_clear_scenery_mousedown(rct_window* w, rct_widgetindex widgetIndex, [[maybe_unused]] rct_widget* widget)
+class CleanSceneryWindow final : public Window
 {
-    switch (widgetIndex)
-    {
-        case WIDX_DECREMENT:
-            // Decrement land tool size, if it stays within the limit
-            gLandToolSize = std::max(MINIMUM_TOOL_SIZE, gLandToolSize - 1);
-
-            // Invalidate the window
-            w->Invalidate();
-            break;
-        case WIDX_INCREMENT:
-            // Increment land tool size, if it stays within the limit
-            gLandToolSize = std::min(MAXIMUM_TOOL_SIZE, gLandToolSize + 1);
-
-            // Invalidate the window
-            w->Invalidate();
-            break;
-    }
-}
+    public:
+        void OnOpen() override
+        {
+            widgets = window_clear_scenery_widgets;
+            enabled_widgets = (1ULL << WIDX_CLOSE) | (1ULL << WIDX_INCREMENT) | (1ULL << WIDX_DECREMENT)
+                | (1ULL << WIDX_PREVIEW) | (1ULL << WIDX_SMALL_SCENERY) | (1ULL << WIDX_LARGE_SCENERY) | (1ULL << WIDX_FOOTPATH);
+            hold_down_widgets = (1ULL << WIDX_INCREMENT) | (1ULL << WIDX_DECREMENT);
+            WindowInitScrollWidgets(this);
+            window_push_others_below(this);
+
+            gLandToolSize = 2;
+            gClearSceneryCost = MONEY64_UNDEFINED;
+
+            gClearSmallScenery = true;
+            gClearLargeScenery = false;
+            gClearFootpath = false;
+        }
+
+        void OnClose() override
+        {
+            if (clear_scenery_tool_is_active())
+                tool_cancel();
+        }
+
+        void OnMouseUp(rct_widgetindex widgetIndex) override
```

Comment:
Go through and add const on pretty much every parameter in every function call of this window. (Everything except OnDraw which does modify its parameter)